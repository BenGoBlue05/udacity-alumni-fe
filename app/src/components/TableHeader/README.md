## TableHeader Component
A component that ...

### Example

```js
<TableHeader />
```


### Other Information
