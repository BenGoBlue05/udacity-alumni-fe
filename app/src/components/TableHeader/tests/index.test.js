import TableHeader from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<TableHeader />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <TableHeader 
      	sortIndex={2}
        sortAscending={true}
        onSort={(e) => e}
        labels={['', 'Title', 'Posted On', 'Status', 'Author', 'Actions']}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
