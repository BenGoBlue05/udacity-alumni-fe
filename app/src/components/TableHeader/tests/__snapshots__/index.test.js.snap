exports[`<TableHeader /> should render with default props 1`] = `
<thead>
  <tr>
    <th>
      <Button
        fill={true}
        onClick={[Function bound _onSort]}
        plain={true}
        type="button">
        <Box
          align="center"
          announce={false}
          direction="row"
          focusable={true}
          justify="start"
          pad={
            Object {
              "between": "small"
            }
          }
          responsive={true}
          tag="div">
          <span />
        </Box>
      </Button>
    </th>
    <th>
      <Button
        fill={true}
        onClick={[Function bound _onSort]}
        plain={true}
        type="button">
        <Box
          align="center"
          announce={false}
          direction="row"
          focusable={true}
          justify="start"
          pad={
            Object {
              "between": "small"
            }
          }
          responsive={true}
          tag="div">
          <span>
            Title
          </span>
        </Box>
      </Button>
    </th>
    <th>
      <Button
        fill={true}
        onClick={[Function bound _onSort]}
        plain={true}
        type="button">
        <Box
          align="center"
          announce={false}
          direction="row"
          focusable={true}
          justify="start"
          pad={
            Object {
              "between": "small"
            }
          }
          responsive={true}
          tag="div">
          <span>
            Posted On
          </span>
          <LinkDown
            a11yTitleId="link-down-title"
            responsive={true} />
        </Box>
      </Button>
    </th>
    <th>
      <Button
        fill={true}
        onClick={[Function bound _onSort]}
        plain={true}
        type="button">
        <Box
          align="center"
          announce={false}
          direction="row"
          focusable={true}
          justify="start"
          pad={
            Object {
              "between": "small"
            }
          }
          responsive={true}
          tag="div">
          <span>
            Status
          </span>
        </Box>
      </Button>
    </th>
    <th>
      <Button
        fill={true}
        onClick={[Function bound _onSort]}
        plain={true}
        type="button">
        <Box
          align="center"
          announce={false}
          direction="row"
          focusable={true}
          justify="start"
          pad={
            Object {
              "between": "small"
            }
          }
          responsive={true}
          tag="div">
          <span>
            Author
          </span>
        </Box>
      </Button>
    </th>
    <th>
      <Button
        fill={true}
        onClick={[Function bound _onSort]}
        plain={true}
        type="button">
        <Box
          align="center"
          announce={false}
          direction="row"
          focusable={true}
          justify="start"
          pad={
            Object {
              "between": "small"
            }
          }
          responsive={true}
          tag="div">
          <span>
            Actions
          </span>
        </Box>
      </Button>
    </th>
  </tr>
</thead>
`;
